### DISCUSSION TITLE
Bug with mode="point" for ROI.get_z_from_dsm()

### DISCUSSION THREAD

**Discussion by youngdjn** [UTokyo-FieldPhenomics-Lab/EasyIDP#71] [General]
Thank you for making this excellent package! It is incredibly useful. I have a use case where I need to use the `mode="point"` option for `ROI.get_z_from_dsm()`, but there appears to be a bug with this option. When I run the following:

```
import easyidp as idp
lotus = idp.data.Lotus()
roi = idp.ROI(lotus.shp, name_field = "plot_id")
roi.get_z_from_dsm(lotus.metashape.dsm, mode="point")
```

I get: 

``
ValueError: all the input arrays must have same number of dimensions, but the array at index 0 has 2 dimension(s) and the array at index 1 has 3 dimension(s)
``

I need the `mode="point"` option because I am trying to extract images of the crowns of individual trees, which have a lot of height variation. When I use with `mode="face"`, the projected ROI does not follow my feature quite right (though it's close, and generally still usable).

<img width="514" alt="image" src="https://user-images.githubusercontent.com/22568494/215301748-4801d4f0-24d9-4152-9787-d432961efe6e.png">

<img width="792" alt="image" src="https://user-images.githubusercontent.com/22568494/215300088-606dd573-bf04-492e-8668-48aacb605f49.png">

(I drew the "pacman" shape because I was hoping the central vertex would track the top of the tree.)

**Comment by HowcanoeWang** [comment]
Thank you for reporting, yes, it is a bug and we have fixed it at ac7b437

Please refer to [Using from the source code](https://easyidp.readthedocs.io/en/latest/install.html#using-from-source-code), to use the latest updates.

And in your case, yes, the "point" mode is the best option, the "face" mode using the mean height of the pacman area, so the tree tip is not the correct place.

But I could imagine, even you get the correct height by "point" mode, the image [DJI_0441] may not the best option, because the location are not in the image center.  Please also refer to [Find the best backward image](https://easyidp.readthedocs.io/en/latest/jupyter/backward_projection.html#Find-the-best-backward-image)

**Reply by youngdjn** [reply]
Thanks for fixing this so quickly! It works great now!

<img width="788" alt="image" src="https://user-images.githubusercontent.com/22568494/215345642-1fcd30b6-4030-40d0-b4a6-4506e09e0ba8.png">

It looks jagged, but this is the expected result, and it's the reason why I made the pacman shape, so that one vertex approximately tracks the treetop to inform the crop.

I agree that image DJI_0441 is not ideal. Here it is just an example of an extreme case. My ultimate goal is to extract the crown of every tree from every photo it appears in, including photos where the view is oblique, and feed the photos through a computer vision model to identify the tree species, using perhaps the modal species ID across all photos. Bad photos with low confidence scores (like DJI_0441 presumably) can be given less weight.
